conda/plugins/types.py

```diff
@@ -24,11 +24,25 @@ class CondaSubcommand(NamedTuple):
 
 class CondaVirtualPackage(NamedTuple):
     """
-    Conda subcommand entry.
+    A conda virtual package.
 
     :param name: Virtual package name (e.g., ``my_custom_os``).
     :param version: Virtual package version (e.g., ``1.2.3``).
     """
 
     name: str
     version: str | None
+
+
+class CondaSolver(NamedTuple):
+    """
+    A conda solver.
+
+    :param name: Subcommand name (e.g., ``conda my-subcommand-name``).
+    :param backend: Callable that will instantiated as the solver backend.
+    """
+    name: str
+    backend: Callable[
+        [list[str]],
+        int | None,
```

Comment:
Doesn't this need to be Sovler object. Looks a lot like the CondaSubCommand type 🤔

Reply:
True, that's weird!